distribution/kubernetes/searcher-statefulset.yaml

```diff
@@ -0,0 +1,94 @@
+apiVersion: apps/v1
+kind: StatefulSet
```

Comment:
Do we really need a statefulset for searchers?

Reply:
Yes, I believe our logic for node/split affinity via rendez-vous hashing relies on stable node IDs.

Reply:
Legit.

Reply:
I'm not sure I understand the sts here either. Does the node/split affinity depend on the presence of previously existing cache on disk? Meaning, if you delete/restart a sts pod the name will stay the same but the emptyDir for cache will be gone. So it would need to repopulate any split cache it had previously regardless?

Reply:
You're right; in this contrived example, stable node IDs between restarts are useless because the `emptyDir` is trashed on restart.

These manifests are intended to serve as a basis for customization. I hope to commit a helm chart to make this easier in the next few weeks.

Reply:
We have already created a helm chart using deployments if you're interested. If I knew where you wanted it I can submit it to you via a PR or just put it somewhere for you guys to get. Its not perfect or completely flushed out but might help.

Reply:
That would be a great start for us. I've just created a [repo](https://github.com/quickwit-oss/helm-charts) for our helm charts and invited you to the Quickwit dev team on GitHub.

How do you handle stable node IDs with deployments?